nav2_rotation_shim_controller/src/nav2_rotation_shim_controller.cpp

```diff
@@ -139,6 +144,42 @@ geometry_msgs::msg::TwistStamped RotationShimController::computeVelocityCommands
   const geometry_msgs::msg::Twist & velocity,
   nav2_core::GoalChecker * goal_checker)
 {
+  // Rotate to goal heading when in goal xy tolerance
+  if (rotate_to_goal_heading_) {
+    std::lock_guard<std::mutex> lock_reinit(mutex_);
+
+    try {
+      geometry_msgs::msg::Pose base_pose = transformPoseToBaseFrame(pose);
+      geometry_msgs::msg::PoseStamped sampled_pt_goal_stamped = getSampledPathGoal();
+      sampled_pt_goal_stamped.header.stamp = current_path_.header.stamp;
+      geometry_msgs::msg::Pose sampled_pt_goal = transformPoseToBaseFrame(sampled_pt_goal_stamped);
+
+      if (nav2_rotation_shim_controller::utils::withinPositionGoalTolerance(
+          goal_checker,
+          base_pose,
+          sampled_pt_goal))
+      {
+        double pose_yaw, pose_pitch, pose_roll;
+        double goal_yaw, goal_pitch, goal_roll;
+        tf2::getEulerYPR(base_pose.orientation, pose_yaw, pose_pitch, pose_roll);
+        tf2::getEulerYPR(sampled_pt_goal.orientation, goal_yaw, goal_pitch, goal_roll);
+
+        double angular_distance_to_heading = std::abs(goal_yaw - pose_yaw);
+        if (angular_distance_to_heading > M_PI) {
```

Comment:
This becomes unnecessary since it would be within -PI to PI

Reply:
It is still necessary to perform the verification that the angle is within -PI to PI after making the difference. if goal `goal_yaw ~= PI` and` pose_yaw ~= -PI`, the resulting angle would be 2PI, forcing the robot to make a full rotation.

Or did I miss something about your advice with the `tf2::getYaw` function ? Is it possible to substract the quaternions directly and call `tf2::getYaw` only once ?